src/api/core/organizations.rs

```
@@ -740,6 +740,10 @@ async fn send_invite(
                     // automatically accept existing users if mail is disabled
                     if !CONFIG.mail_enabled() && !user.password_hash.is_empty() {
                         user_org_status = UserOrgStatus::Accepted as i32;
+
+                        if OrgPolicy::org_is_reset_password_auto_enroll(&org_id, &mut conn).await {
+                            err!("With disabled mailing and enabled auto-enrollment-feature of reset-password-policy it's not possible to invite existing users");
+                        }
```

> And, also check if the policy is enabled when an admin tries to use this feature, mail is still enabled, and not turned of afterwards. Maybe even prevent this policy from being active in that case?
That way we can prevent issue reports, and you can return an error message stating mail has to be enabled.

I added a check that is preventing existing user invitation to current org if auto-enrollment is enabled but mailing is disabled.
This state can be reached if mailing was enabled while setting up the policy and disabling mailing afterwards.
I think this was what you ment.

---

I put the check aswell for non existing users if they get invited. They habe the same restrictions.

---

I refactored the stuff I did in my latest commit

---

While I still have not looked at this yet, but could you fix the clippy warnings `cargo clippy --features sqlite,mysql,postgresql` ? 
You can also install pre-commit and enable it for this repo so all checks are done before you commit :).

---

If I should do a rebase to current main, please tell me